arangod/Aql/OptimizerRules.cpp

```diff
@@ -9275,3 +9275,18 @@ void arangodb::aql::joinIndexNodesRule(Optimizer* opt,
 
   opt->addPlan(std::move(plan), rule, modified);
 }
+
+void arangodb::aql::removeUnnecessaryProjections(
+    Optimizer* opt, std::unique_ptr<ExecutionPlan> plan,
+    OptimizerRule const& rule) {
+  containers::SmallVector<ExecutionNode*, 8> nodes;
+  plan->findNodesOfType(nodes, {EN::INDEX, EN::ENUMERATE_COLLECTION}, true);
+
+  bool modified = false;
+  for (auto* n : nodes) {
+    modified = true;
```

Comment:
This rule will now show up in every plan that uses an IndexNode or an EnumerateCollectionNode, even if the projections were not changed. I think this is not ideal.
Only those optimizer rules should show up that actually modify the plan. We can make recalculateProjections() return a boolean value, indicating whether projections were changed.
In order to easily figure out if projections were changed, we could make the nodes create a hash of their projections, which can hash whatever is in `_projections`, and optionally `_filterProjections`, before and after the call to `recalculateProjections()`.